src/Native/configure.cmake

```diff
@@ -317,6 +317,14 @@ check_symbol_exists(
     "sys/syslimits.h"
     HAVE_OPEN_MAX)
 
+if (CMAKE_SYSTEM_NAME STREQUAL Linux)
```

Comment:
Why are you guarding this behind "if linux"?  Presumably FreeBSD and/or OS X either do, or don't, have heimntlm; and if so it'll just work.  If not, you need a second functionality-based test.